### ISSUE TITLE
AssertionError at Predef.scala:156 (Polynomial.scala:771)

### ISSUE THREAD

**Issue by rainoftime** [uuverifiers/eldarica#21]
Hi , for the following instance,
[771.txt](https://github.com/uuverifiers/eldarica/files/4828854/771.txt)

Eldarica nightly (http://philipp.ruemmer.org/eldarica-bin-nightly.zip)
throws an assertion error
~~~~
Theories: GroebnerMultiplication
Exception in thread "main" java.lang.AssertionError: assertion failed
        at scala.Predef$.assert(Predef.scala:156)
        at ap.theories.nia.Basis.add(Polynomial.scala:771)
        at ap.theories.nia.Basis.simplify(Polynomial.scala:940)
        at ap.theories.nia.GroebnerMultiplication$$anon$1$$anonfun$4.apply(GroebnerMultiplication.scala:362)
        at ap.theories.nia.GroebnerMultiplication$$anon$1$$anonfun$4.apply(GroebnerMultiplication.scala:345)
        at ap.util.LRUCache.apply(LRUCache.scala:37)
        at ap.theories.nia.GroebnerMultiplication$$anon$1.ap$theories$nia$GroebnerMultiplication$$anon$$handleGoalAux(GroebnerMultiplication.scala:345)
        at ap.theories.nia.GroebnerMultiplication$$anon$1.handleGoal(GroebnerMultiplication.scala:212)
        at ap.proof.theoryPlugins.PluginTask.apply(Plugin.scala:392)
        at ap.proof.goal.Goal.step(Goal.scala:395)
        at ap.proof.ModelSearchProver.ap$proof$ModelSearchProver$$findModel(ModelSearchProver.scala:470)
        at ap.proof.ModelSearchProver.ap$proof$ModelSearchProver$$findModel(ModelSearchProver.scala:530)
        at ap.proof.ModelSearchProver.ap$proof$ModelSearchProver$$findModel(ModelSearchProver.scala:476)
        at ap.proof.ModelSearchProver$IncProverImpl.checkValidityDir(ModelSearchProver.scala:1079)
        at ap.proof.ModelSearchProver$IncProverImpl.checkValidityDir(ModelSearchProver.scala:1069)
        at ap.proof.ModelSearchProver$IncProverImpl.checkValidity(ModelSearchProver.scala:1057)
        at lazabs.horn.bottomup.HornPredAbs$$anonfun$isValid$1.apply$mcZ$sp(HornPredAbs.scala:703)
        at lazabs.horn.bottomup.HornPredAbs$$anonfun$isValid$1.apply(HornPredAbs.scala:694)
        at lazabs.horn.bottomup.HornPredAbs$$anonfun$isValid$1.apply(HornPredAbs.scala:694)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
        at ap.util.Timeout$.withChecker(Timeout.scala:44)
        at lazabs.horn.bottomup.HornPredAbs.isValid(HornPredAbs.scala:694)
        at lazabs.horn.bottomup.HornPredAbs$$anonfun$genEdge$1.apply(HornPredAbs.scala:1430)
        at lazabs.horn.bottomup.HornPredAbs$$anonfun$genEdge$1.apply(HornPredAbs.scala:1421)
        at lazabs.horn.bottomup.Hasher.scope(Hasher.scala:356)
        at lazabs.horn.bottomup.HornPredAbs.genEdge(HornPredAbs.scala:1421)
        at lazabs.horn.bottomup.HornPredAbs.liftedTree1$1(HornPredAbs.scala:870)
        at lazabs.horn.bottomup.HornPredAbs.<init>(HornPredAbs.scala:869)
        at lazabs.horn.bottomup.InnerHornWrapper$$anonfun$24.apply(HornWrapper.scala:371)
        at lazabs.horn.bottomup.InnerHornWrapper$$anonfun$24.apply(HornWrapper.scala:366)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
        at scala.Console$.withOut(Console.scala:65)
        at lazabs.horn.bottomup.InnerHornWrapper.<init>(HornWrapper.scala:366)
        at lazabs.horn.bottomup.HornWrapper$$anonfun$9.apply(HornWrapper.scala:228)
        at lazabs.horn.bottomup.HornWrapper$$anonfun$9.apply(HornWrapper.scala:230)
        at lazabs.ParallelComputation$.apply(ParallelComputation.scala:46)
        at lazabs.horn.bottomup.HornWrapper.<init>(HornWrapper.scala:227)
        at lazabs.horn.Solve$.apply(Solve.scala:81)
        at lazabs.Main$.doMain(Main.scala:586)
        at lazabs.Main$.main(Main.scala:267)
        at lazabs.Main.main(Main.scala)

~~~~

**Comment by pruemmer** [comment]
This problem I'm still looking into!

**Comment by pruemmer** [comment]
I have updated he snapshot on philipp.ruemmer.org now, which I believe
fixes this problem.
http://philipp.ruemmer.org/eldarica-bin-nightly.zip

**Comment by rainoftime** [comment]
Fixed at commit d8af1335
